parlai/mturk/core/dev/agents.py

```diff
@@ -298,52 +243,21 @@ def get_connection_id(self):
         """Returns an appropriate connection_id for this agent"""
         return "{}_{}".format(self.worker_id, self.assignment_id)
 
-    def log_reconnect(self):
-        """Log a reconnect of this agent """
-        shared_utils.print_and_log(
-            logging.DEBUG,
-            'Agent ({})_({}) reconnected to {} with status {}'.format(
-                self.worker_id,
-                self.assignment_id,
-                self.conversation_id,
-                self.get_status(),
-            ),
-        )
-
-    def get_inactive_command_data(self):
-        """Get appropriate inactive command data to respond to a reconnect"""
-        text, command = self.state.get_inactive_command_text()
-        return {
-            'text': command,
-            'inactive_text': text,
-            'conversation_id': self.conversation_id,
-            'agent_id': self.worker_id,
-        }
-
-    def wait_for_status(self, desired_status):
-        """Suspend a thread until a particular assignment state changes
-        to the desired state
-        """
-        while True:
-            if self.get_status() == desired_status:
-                return True
-            if self.is_final():
-                return False
-            time.sleep(shared_utils.THREAD_SHORT_SLEEP)
-
     def is_in_task(self):
-        """Use conversation_id to determine if an agent is in a task"""
-        if self.conversation_id:
-            return 't_' in self.conversation_id
-        return False
+        """Simple check for an agent being in a task"""
+        return self.get_status() == AssignState.STATUS_IN_TASK
 
     def observe(self, msg):
         """Send an agent a message through the mturk_manager"""
-        self.mturk_manager.send_message(self.worker_id, self.assignment_id, msg)
+        if 'message_id' not in msg:
+            msg['message_id'] = str(uuid.uuid4())
+        self.state.append_message(msg)
```

Comment:
I made the agent responsible for its own message tracking, as having this happen in various other classes didn't make sense when all messages were coming through here at some point.